### ISSUE TITLE
AWS ECS Entrypoint container support required - ECS Task failing 

### ISSUE THREAD

**Issue by prashantchawla2** [aws/aws-for-fluent-bit#39]
We need an entrypoint in the ECS Task Definition for this container. But when we try to add it fails because we don't know what "path" and "command" that needs to go into the Entrypoint. Because there is no entrypoint our security solution with Twistlock Defender is not working, because Twistlock Defender works as a Sidecar and puts in an entry in the Entrypoint. For other applications it is working, but for this specific container "amazon/aws-for-fluent-bit:latest" it is not working. We don't know what is a valid "path" and "command" that needs to go in there.

"entryPoint": [
                "/twistlock/fargate_defender.sh",
                "fargate",
                "entrypoint",
                "/entrypoint.sh",
                "run"
            ],

**Comment by PettitWesley** [comment]
@prashantchawla2 You can see the entrypoint command at the end here: https://github.com/aws/aws-for-fluent-bit/blob/master/entrypoint.sh

And the Dockerfile invokes it here: https://github.com/aws/aws-for-fluent-bit/blob/master/Dockerfile#L109

**Comment by hossain-rayhan** [comment]
Hi @prashantchawla2 , hope your issue was resolved. We are closing this issue. Feel free to reopen if you still need any help.

**Comment by PettitWesley** [comment]
@prashantchawla2 We have spoken to Twistlock and determined the solution.

### Step 1: Start with Container Definition with Fluent Bit which has proper entrypoint

Take your task definition and set the Fluent Bit container definition to be:

```
"entryPoint": [
"/entrypoint.sh",
],
```

Do not worry about the twistlock defender, that will be added in the next step. 

### Step 2: Use Twistlock tools to convert Task Definition

I was told by twistlock that they have a tool for taking a task definition, adding the defender, and correctly setting the entrypoint. The only requirement is that the input task definition has entrypoint's defined (w/o twistlock defender). Hence start with the entrypoint above that has no twistlock integration.

Then follow the twistlock documentation: https://docs.paloaltonetworks.com/prisma/prisma-cloud/19-11/prisma-cloud-compute-edition-admin/runtime_defense/fargate.html

Basically, you don't need to try to figure out how that twistlock defender script works- their tool will help you do it once you have the entrypoint defined.

**Comment by PettitWesley** [comment]
Closing since we provided the docs from twistlock
